src/components/modebar/buttons.js

```diff
@@ -533,8 +533,18 @@ function handleGeo(gd, ev) {
 
         if(attr === 'zoom') {
             var scale = geoLayout.projection.scale;
+            var minscale = geoLayout.projection.minscale;
+            var maxscale = geoLayout.projection.maxscale;
```

Comment:
Now we need to handle 
```js
if(maxscale === null) maxscale = Infinity;
```
or
```js
if(maxscale === -1) maxscale = Infinity;
```
if we decide to use `-1` instead of `null` which is not a `number`.
@alexcjohnson your idea on which one is better?

Reply:
With e7229479d7707fad5e7e319c7997130912e7835b I changed the default to `-1` instead of `null`. Scales with less than 0 make no sense here so this should be a valid solution.

Reply:
I suppose `-1` as a default is OK if we really need a "no max" value - it's better than `null` which behaves oddly in `relayout`.

But I wonder if we can't pick a decent default that's more zoomed in than is generally ever useful. I have occasionally gotten in a situation by accidentally scroll zooming so far in that everything freezes up because the SVG engine is having trouble drawing all the off-screen elements. If we cover 99% of cases and avoid this pitfall I'd say that's a reasonable tradeoff, and avoids making a coded value. Something like `10,000`?

Similarly, is there a default we could use for minscale, to avoid zooming the whole map away to a point? `0.1` perhaps?

Reply:
> Something like 10,000?
> Similarly, is there a default we could use for minscale, to avoid zooming the whole map away to a point? 0.1 perhaps?

Makes sense!
(I'll have a look tomorrow. It's getting late her in europe :sleeping:)